honeycombio/resource_board_test.go

```diff
@@ -0,0 +1,131 @@
+package honeycombio
+
+import (
+    "errors"
+    "fmt"
+    "testing"
+
+    "github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+    "github.com/hashicorp/terraform-plugin-sdk/terraform"
+    honeycombio "github.com/kvrhdn/go-honeycombio"
+    "github.com/stretchr/testify/assert"
+)
+
+func TestAccHoneycombioBoard_basic(t *testing.T) {
+    dataset := testAccDataset()
+
+    resource.Test(t, resource.TestCase{
+        PreCheck:  func() { testAccPreCheck(t) },
+        Providers: testAccProviders,
+        Steps: []resource.TestStep{
+            {
+                Config: testAccBoardConfig(dataset),
+                Check: resource.ComposeTestCheckFunc(
+                    testAccCheckBoardExists(t, "honeycombio_board.test"),
+                ),
+            },
+        },
+    })
+}
+
+func testAccBoardConfig(dataset string) string {
+    return fmt.Sprintf(`
+data "honeycombio_query" "test" {
+    count = 2
+    calculation {
+        op     = "AVG"
+        column = "duration_ms"
+    }
+    filter {
+        op     = ">"
+        column = "duration_ms"
+        value  = count.index 
+    }
+}
+
+resource "honeycombio_board" "test" {
+    name  = "Test board from terraform-provider-honeycombio"
+    style = "list"
+     
+    query {
+        caption = "test query 0"
+        dataset = "%s"
+        query_json = data.honeycombio_query.test[0].json
+    }
+      
+    query {
+        caption = "test query 1"
+        dataset = "%s"
+        query_json = data.honeycombio_query.test[1].json
+    }
+
+}`, dataset, dataset)
```

Comment:
This part look reasonable to you?

Reply:
Yeah, I think this is great. This seems intuitive and easy to use.
I had put something similar here: https://github.com/kvrhdn/terraform-provider-honeycombio/issues/3#issue-669230480